ui/app/AppLayouts/Wallet/controls/ManageTokensCommunityTag.qml

```diff
@@ -47,16 +53,59 @@ Control {
             Layout.preferredHeight: visible ? 16 : 0
             asset.width: 16
             asset.height: 16
-            asset.isImage: true
-            visible: !!asset.source
+            visible: root.full && !!asset.source
+            asset.name: !!root.communityImage ? root.communityImage : "help"
+            asset.isImage: !!root.communityImage
         }
-        StatusBaseText {
+
+        RowLayout {
+            spacing: 2
             Layout.fillWidth: true
-            font.pixelSize: Style.current.tertiaryTextFontSize
-            font.weight: Font.Medium
-            text: root.text
-            elide: Text.ElideRight
-            color: enabled ? Theme.palette.directColor1 : Theme.palette.baseColor1
+
+            StatusBaseText {
+                font.pixelSize: Style.current.tertiaryTextFontSize
+                font.weight: Font.Medium
+                text:  {
+                    if (d.unknown) {
+                        if (communityNameToolTip.visible) {
+                            if (!root.full) {
+                                return StatusQUtils.Utils.elideAndFormatWalletAddress(root.communityName)
+                            }
+                            return qsTr("Community %1").arg(StatusQUtils.Utils.elideAndFormatWalletAddress(root.communityName))
+                        }
+                        if (!root.full) {
+                            return qsTr("Unknown")
+                        }
+
+                        return qsTr("Unknown community")
+                    }
+                    
+                    return root.communityName
+                }
+                elide: Text.ElideRight
+                color: enabled ? Theme.palette.directColor1 : Theme.palette.baseColor1                
+            }
+
+            CopyToClipBoardButton {
+                visible: d.unknown && root.hovered
+                icon.height: Style.current.tertiaryTextFontSize
+                icon.width: Style.current.tertiaryTextFontSize
+                icon.color: Theme.palette.directColor1
+                Layout.preferredWidth: 16
+                Layout.preferredHeight: 16
+                color: Style.current.transparent
+                textToCopy: root.communityName
+                onCopyClicked: {
+                    root.walletStore.copyToClipboard(textToCopy)
```

Comment:
@caybro how do we deal with this `copyToClipboard` calls in other cases? I see we have `Utils.copyToClipboard` that could save us here, passing the store.. WDYT?

Reply:
that would be nice if i could avoid to pass the store

Reply:
Yeah, just use `Utils.copyToClipboard(text)`